### ISSUE TITLE
[firefox-git] Permission denied: '/usr/local/lib/python2.7/dist-packages/urllib3-1.22.dist-info'

### ISSUE THREAD

**Issue by jankeromnes** [JanitorTechnology/dockerfiles#101]
- Failing command: `cd firefox docker login -u $DOCKER_USER -p $DOCKER_PASS docker pull janx/ubuntu-dev docker build -t janx/firefox -f firefox-git.dockerfile . docker push janx/firefox`
- Exit code: `2`
- Output:

```
/libpulse0_1%3a8.0-0ubuntu3.4_amd64.deb ...

Unpacking libpulse0:amd64 (1:8.0-0ubuntu3.4) ...

Selecting previously unselected package libpulse-mainloop-glib0:amd64.

Preparing to unpack .../libpulse-mainloop-glib0_1%3a8.0-0ubuntu3.4_amd64.deb ...

Unpacking libpulse-mainloop-glib0:amd64 (1:8.0-0ubuntu3.4) ...

Selecting previously unselected package libpulse-dev:amd64.

Preparing to unpack .../libpulse-dev_1%3a8.0-0ubuntu3.4_amd64.deb ...

Unpacking libpulse-dev:amd64 (1:8.0-0ubuntu3.4) ...

Selecting previously unselected package libxt-dev:amd64.

Preparing to unpack .../libxt-dev_1%3a1.1.5-0ubuntu1_amd64.deb ...

Unpacking libxt-dev:amd64 (1:1.1.5-0ubuntu1) ...

Selecting previously unselected package python-dbus.

Preparing to unpack .../python-dbus_1.2.0-3_amd64.deb ...

Unpacking python-dbus (1.2.0-3) ...

Selecting previously unselected package python-gi.

Preparing to unpack .../python-gi_3.20.0-0ubuntu1_amd64.deb ...

Unpacking python-gi (3.20.0-0ubuntu1) ...

Selecting previously unselected package yasm.

Preparing to unpack .../yasm_1.3.0-2_amd64.deb ...

Unpacking yasm (1.3.0-2) ...

Selecting previously unselected package libiw-dev:amd64.

Preparing to unpack .../libiw-dev_30~pre9-8ubuntu1_amd64.deb ...

Unpacking libiw-dev:amd64 (30~pre9-8ubuntu1) ...

Processing triggers for libc-bin (2.23-0ubuntu9) ...

Processing triggers for man-db (2.7.5-1) ...

Processing triggers for libglib2.0-0:amd64 (2.48.2-0ubuntu1) ...

Setting up libjson-c2:amd64 (0.11-4ubuntu2) ...

Setting up libasyncns0:amd64 (0.8-5build1) ...

Setting up libogg0:amd64 (1.3.2-1) ...

Setting up libpcrecpp0v5:amd64 (2:8.38-3.1) ...

Setting up libiw30:amd64 (30~pre9-8ubuntu1) ...

Setting up libgirepository-1.0-1:amd64 (1.46.0-3ubuntu1) ...

Setting up gir1.2-glib-2.0:amd64 (1.46.0-3ubuntu1) ...

Setting up autotools-dev (20150820.1) ...

Setting up dbus-x11 (1.10.6-1ubuntu3.3) ...

Setting up intltool-debian (0.35.0+20060710.4) ...

Setting up po-debconf (1.0.19) ...

Setting up libarchive-zip-perl (1.56-2) ...

Setting up libfile-stripnondeterminism-perl (0.015-1) ...

Setting up gconf2 (3.2.6-3ubuntu6) ...

Setting up gir1.2-atk-1.0 (2.18.0-1) ...

Setting up gir1.2-atspi-2.0 (2.18.3-4ubuntu1) ...

Setting up gir1.2-freedesktop:amd64 (1.46.0-3ubuntu1) ...

Setting up gir1.2-gconf-2.0 (3.2.6-3ubuntu6) ...

Setting up gir1.2-gdkpixbuf-2.0:amd64 (2.32.2-1ubuntu1.3) ...

Setting up libgtk2.0-common (2.24.30-1ubuntu1.16.04.2) ...

Setting up libpangoxft-1.0-0:amd64 (1.38.1-1) ...

Setting up gir1.2-pango-1.0:amd64 (1.38.1-1) ...

Setting up libgtk2.0-0:amd64 (2.24.30-1ubuntu1.16.04.2) ...

Setting up gir1.2-gtk-2.0 (2.24.30-1ubuntu1.16.04.2) ...

Setting up gir1.2-gtk-3.0:amd64 (3.18.9-1ubuntu3.3) ...

Setting up gir1.2-notify-0.7 (0.7.6-2svn1) ...

Setting up libasound2-dev:amd64 (1.1.0-0ubuntu1) ...

Setting up libglib2.0-bin (2.48.2-0ubuntu1) ...

Setting up libpcre16-3:amd64 (2:8.38-3.1) ...

Setting up libpcre32-3:amd64 (2:8.38-3.1) ...

Setting up libpcre3-dev:amd64 (2:8.38-3.1) ...

Setting up pkg-config (0.29.1-0ubuntu1) ...

Setting up libglib2.0-dev (2.48.2-0ubuntu1) ...

Setting up libdbus-1-dev:amd64 (1.10.6-1ubuntu3.3) ...

Setting up libxi-dev (2:1.7.6-1) ...

Setting up x11proto-record-dev (1.14.2-1) ...

Setting up libxtst-dev:amd64 (2:1.2.2-1) ...

Setting up libatspi2.0-dev (2.18.3-4ubuntu1) ...

Setting up libatk-bridge2.0-dev:amd64 (2.18.1-2ubuntu1) ...

Setting up libatk1.0-dev (2.18.0-1) ...

Setting up libcairo-script-interpreter2:amd64 (1.14.6-1) ...

Setting up libpng12-dev:amd64 (1.2.54-1ubuntu1) ...

Setting up libfreetype6-dev:amd64 (2.6.1-0.1ubuntu2.3) ...

Setting up libfontconfig1-dev:amd64 (2.11.94-0ubuntu1.1) ...

Setting up x11proto-render-dev (2:0.11.1-2) ...

Setting up libxrender-dev:amd64 (1:0.9.9-0ubuntu1) ...

Setting up libice-dev:amd64 (2:1.0.9-1) ...

Setting up libsm-dev:amd64 (2:1.2.2-1) ...

Setting up libpixman-1-dev (0.33.6-1) ...

Setting up libxcb-shm0-dev:amd64 (1.11.1-1ubuntu1) ...

Setting up libcairo2-dev (1.14.6-1) ...

Setting up libdbus-glib-1-dev (0.106-1) ...

Setting up libwayland-bin (1.12.0-1~ubuntu16.04.2) ...

Setting up libwayland-dev:amd64 (1.12.0-1~ubuntu16.04.2) ...

Setting up libmircore-dev:amd64 (0.26.3+16.04.20170605-0ubuntu1.1) ...

Setting up libprotobuf9v5:amd64 (2.6.1-1.3) ...

Setting up libprotobuf-dev:amd64 (2.6.1-1.3) ...

Setting up libxkbcommon-dev (0.5.0-1ubuntu2) ...

Setting up libmircommon-dev:amd64 (0.26.3+16.04.20170605-0ubuntu1.1) ...

Setting up libmircookie2:amd64 (0.26.3+16.04.20170605-0ubuntu1.1) ...

Setting up libmircookie-dev:amd64 (0.26.3+16.04.20170605-0ubuntu1.1) ...

Setting up libmirclient-dev:amd64 (0.26.3+16.04.20170605-0ubuntu1.1) ...

Setting up libegl1-mesa-dev:amd64 (17.0.7-0ubuntu0.16.04.2) ...

Setting up libepoxy-dev:amd64 (1.3.1-1ubuntu0.16.04.2) ...

Setting up libflac8:amd64 (1.3.1-4) ...

Setting up libgconf2-dev (3.2.6-3ubuntu6) ...

Setting up libgconf2-doc (3.2.6-3ubuntu6) ...

Setting up libgdk-pixbuf2.0-dev (2.32.2-1ubuntu1.3) ...

Setting up libgmpxx4ldbl:amd64 (2:6.1.0+dfsg-2) ...

Setting up libgmp-dev:amd64 (2:6.1.0+dfsg-2) ...

Setting up libharfbuzz-gobject0:amd64 (1.0.1-1ubuntu0.1) ...

Setting up libharfbuzz-dev (1.0.1-1ubuntu0.1) ...

Setting up libxft-dev (2.3.2-1) ...

Setting up libpango1.0-dev (1.38.1-1) ...

Setting up x11proto-xinerama-dev (1.2.1-2) ...

Setting up libxinerama-dev:amd64 (2:1.1.3-1) ...

Setting up x11proto-randr-dev (1.5.0-1) ...

Setting up libxrandr-dev:amd64 (2:1.5.0-1) ...

Setting up libxcursor-dev:amd64 (1:1.1.14-1) ...

Setting up x11proto-composite-dev (1:0.4.2-2) ...

Setting up libxcomposite-dev (1:0.4.4-1) ...

Setting up nettle-dev (3.2-1ubuntu0.16.04.1) ...

Setting up libgtk-3-dev:amd64 (3.18.9-1ubuntu3.3) ...

Setting up libgtk2.0-bin (2.24.30-1ubuntu1.16.04.2) ...

Setting up libgtk2.0-dev (2.24.30-1ubuntu1.16.04.2) ...

Setting up libsys-hostname-long-perl (1.5-1) ...

Setting up libmail-sendmail-perl (0.79.16-1) ...

Setting up libnotify-dev (0.7.6-2svn1) ...

Setting up libvorbis0a:amd64 (1.3.5-3) ...

Setting up libvorbisenc2:amd64 (1.3.5-3) ...

Setting up libsndfile1:amd64 (1.0.25-10ubuntu0.16.04.1) ...

Setting up libpulse0:amd64 (1:8.0-0ubuntu3.4) ...

Setting up libpulse-mainloop-glib0:amd64 (1:8.0-0ubuntu3.4) ...

Setting up libpulse-dev:amd64 (1:8.0-0ubuntu3.4) ...

Setting up libxt-dev:amd64 (1:1.1.5-0ubuntu1) ...

Setting up python-dbus (1.2.0-3) ...

Remove stale byte-compiled files...

Setting up python-gi (3.20.0-0ubuntu1) ...

Setting up yasm (1.3.0-2) ...

Setting up libiw-dev:amd64 (30~pre9-8ubuntu1) ...

Setting up dh-strip-nondeterminism (0.015-1) ...

Setting up debhelper (9.20160115ubuntu3) ...

Processing triggers for libc-bin (2.23-0ubuntu9) ...

Executing as root: sudo apt-get install -y autoconf2.13 build-essential ccache python-dev python-pip python-setuptools unzip uuid zip
Executing as root: sudo apt-get install -y libasound2-dev libcurl4-openssl-dev libdbus-1-dev libdbus-glib-1-dev libgconf2-dev libgtk-3-dev libgtk2.0-dev libiw-dev libnotify-dev libpulse-dev libx11-xcb-dev libxt-dev mesa-common-dev python-dbus xvfb yasm
Your version of Mercurial (4.3.2) is sufficiently modern.
Your version of Python (2.7.12) is new enough.
Your version of Rust (1.21.0) is new enough.
Rust supports x86_64-unknown-linux-gnu targets.

Source code can be obtained by running

    hg clone https://hg.mozilla.org/mozilla-unified

Or, if you prefer Git, you should install git-cinnabar, and follow the
instruction here to clone from the Mercurial repository:

    https://github.com/glandium/git-cinnabar/wiki/Mozilla:-A-git-workflow-for-Gecko-development

Or, if you really prefer vanilla flavor Git:

    git clone https://github.com/mozilla/gecko-dev.git


Your system should be ready to build Firefox for Desktop!

 ---> bd8e85b04961
Removing intermediate container affff74fc4c9
Step 4/15 : RUN pip install requests
 ---> Running in 3f4cfbf4eb19
Collecting requests
  Downloading requests-2.18.4-py2.py3-none-any.whl (88kB)
Collecting urllib3<1.23,>=1.21.1 (from requests)
  Downloading urllib3-1.22-py2.py3-none-any.whl (132kB)
Collecting idna<2.7,>=2.5 (from requests)
  Downloading idna-2.6-py2.py3-none-any.whl (56kB)
Collecting chardet<3.1.0,>=3.0.2 (from requests)
  Downloading chardet-3.0.4-py2.py3-none-any.whl (133kB)
Collecting certifi>=2017.4.17 (from requests)
  Downloading certifi-2017.7.27.1-py2.py3-none-any.whl (349kB)
Installing collected packages: urllib3, idna, chardet, certifi, requests
Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/local/lib/python2.7/dist-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/usr/local/lib/python2.7/dist-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/usr/local/lib/python2.7/dist-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/local/lib/python2.7/dist-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/usr/local/lib/python2.7/dist-packages/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/usr/local/lib/python2.7/dist-packages/pip/wheel.py", line 316, in clobber
    ensure_dir(destdir)
  File "/usr/local/lib/python2.7/dist-packages/pip/utils/__init__.py", line 83, in ensure_dir
    os.makedirs(path)
  File "/usr/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/urllib3-1.22.dist-info'
The command '/bin/sh -c pip install requests' returned a non-zero code: 2
Exited with code 2
```

**Comment by jankeromnes** [comment]
To fix this, we could install `pip` for `user` instead of system-wide, e.g. with:

    echo "[install]\nuser=1" >> /home/user/.config/pip/pip.conf

See https://pip.pypa.io/en/stable/user_guide/#configuration. Thank you for the tip @Martiusweb!

**Comment by jankeromnes** [comment]
I think this was fixed by d25a150613cb02c32a1d6ef97ba1a4beaeb3494c or a follow-up.
